rfcs/BFDv2/BFDv2 RFC.md

```diff
@@ -0,0 +1,235 @@
+# RFC Proposal
+
+[RFC Proposal]: #rfc-proposal
+
+* RFC Proposal ID: `0000-bfd-v2-fhir-r4-carin`
+* Start Date: TBD
+* RFC PR: TBD
+* JIRA Ticket(s):
+* [BFD-285](https://jira.cms.gov/browse/BFD-285)
+
+
+This RFC proposes that our team migrate from BFD v1 platform currently on FHIR STU3 to BFD v2 platform supporting the first normative FHIR release, R4 and implement CARIN BB profiles.
+
+## Table of Contents
+[Table of Contents]: #table-of-contents
+
+* [RFC Proposal](#rfc-proposal)
+* [Table of Contents](#table-of-contents)
+* [Motivation](#motivation)
+* [Proposed Solution](#proposed-solution)
+	* [Detailed Design](#detailed-design)
+		* [HAPI Version](#hapi-version)
+		* [HAPI Data Model](#hapi-data-model)
+		* [Maven Dependency BFDv2](#maven-dependency-bfdv2)
+		* [FHIR R4 Server](#fhir-r4-server)
+		* [Spring Configuration](#spring-configuration)
+		* [BFDv2 Resource Providers](#bfdv2-resource-providers)
+	* [Current State](#current-state)
+	* [Future State](#future-state)
+	* [Differential View](#differential-view)
+	* [Unresolved Questions](#unresolved-questions)
+	* [Drawbacks](#proposed-solution-drawbacks)
+	* [Notable Alternatives](#proposed-solution-notable-alternatives)
+* [Prior Art](#prior-art)
+* [Future Possibilities](#future-possibilities)
+* [Addendums](#addendums)
+
+
+## Motivation
+[Motivation]: #motivation
+
+BFDv2 will make BFD's FHIR resources increase the system's conformance with FHIR R4 and industry alliance like CARIN.
+This will help BFD move its platform to use the first normative release of FHIR (R4). This will also make integration and 
+interoperability with all our peer partners seamless by enabling to provide FHIR-R4 validated and conformant resources
+for Patient, Coverage and Explanation Of Benefits (EOB's).
+
+
+## Proposed Solution
+[Proposed Solution]: #proposed-solution
+
+BFD proposes to provide a new v2 endpoint URL (future state) and will make no structural or architectural change to the v1 endpoint (current state). The goal is for partners to not have to make any changes in their existing integration points with BFD.
+
+**BFD v2 WILL NOT be backward compatible with BFD v1 and Peer Parrtners will not be able to use their existing v1 connectors/interfaces/integration points for v2 consumption.**
+
+![](./BFDv1_v2.PNG)
+
+BFD v2 endpoint will support the following FHIR resources:  
+
+* Patient (BFD v2, FHIR r4)
+* Coverage (BFD v2, FHIR r4)
+* ExplanationOfBenefit (BFD v2, FHIR r4)
+
+An idicative v2 endpoint is below:  
+
+https://localhost:1337/v2/fhir/Patient?_id=-19990000000001&_format=json
+
+Link to FHIR R4 and CARIN BB Profiles is below:  
+
+FHIR R4 (US-Core, Normative release):
+http://hl7.org/fhir/R4
+
+
+CARIN BB Profiles:
+http://build.fhir.org/ig/HL7/carin-bb
+
+### Detailed Design
+
+[Detailed Design]: #detailed-design

```

Comment:
Could you use this section to detail the technical 'nuts and bolts' to how we intend to implement two versions?